File: pset1/cash/cash.c
// Returns minimum number of coins with which that change can be made
// NataliaSaz problem set 1-3 "Cash" less comfortable level
#include <cs50.h>
#include <stdio.h>
#include <math.h>
float get_positive_float(string prompt);
void print_coins(float change);

int main(void)
{
    float i = get_positive_float("Change: $");
    print_coins(i);
}

// Prompt user for change
float get_positive_float(string prompt)
{
    float n;
    do
    {
        n = get_float("%s", prompt);
    }
    while (n <= 0);
    return n;
}

// Calculates minimum number of coins
void print_coins(float change)
{
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;
    int cents = round(change * 100);
    int qty_of_quarters = cents / quarter;
    int qty_of_dimes = (cents - qty_of_quarters * quarter) / dime;
    int qty_of_nickels = (cents - qty_of_quarters * quarter - qty_of_dimes * dime) / nickel;
    int qty_of_pennies = (cents - qty_of_quarters * quarter - qty_of_dimes * dime - qty_of_nickels * nickel) / penny;
    int coins = qty_of_quarters + qty_of_dimes + qty_of_nickels + qty_of_pennies;
    printf("Number of coins: %i\n", coins);
}

File: pset6/cash/cash.py
# Returns minimum number of coins with which that change can be made


from cs50 import get_float


def main():
    i = get_positive_float("Change: $ ")
    print_coins(i)

# Prompt user for change


def get_positive_float(prompt):
    while True:
        n = get_float(prompt)
        if n > 0:
            break
    return n

# Count how many coins will be given in change


def print_coins(change):
    quarter = 25
    dime = 10
    nickel = 5
    penny = 1
    quarters = int(change * 100 / quarter)
    dimes = int((change * 100 - quarter * quarters)/dime)
    nickels = int((change * 100 - quarter * quarters - dimes * dime)/nickel)
    pennies = int((change * 100 - quarter * quarters - dimes * dime - nickel * nickels)/penny)
    # Total coins

    print(quarters + dimes + nickels + pennies)


if __name__ == "__main__":
    main()